packages/loopring_v3.js/src/types.ts

```diff
@@ -0,0 +1,188 @@
+import BN = require("bn.js");
+import { SparseMerkleTree } from "./SparseMerkleTree";
+
+export enum BlockType {
+    RING_SETTLEMENT = 0,
+    DEPOSIT,
+    ONCHAIN_WITHDRAWAL,
+    OFFCHAIN_WITHDRAWAL,
+    ORDER_CANCELLATION
+  }
+
+  export enum BlockState {
+    NEW = 0,
+    COMMITTED,
+    VERIFIED,
+    FINALIZED,
+  }
+
+  export enum ForgeMode {
+    AUTO_UPGRADABLE = 0,
+    MANUAL_UPGRADABLE,
+    PROXIED,
+    NATIVE,
+  }
+
+  export interface Block {
+    blockIdx: number;
+
+    blockType: BlockType;
+    blockSize: number;
+    blockVersion: number;
+    data: string;
+    offchainData: string;
+
+    operator: string;
+
+    blockState: BlockState;
+
+    merkleRoot: string;
+
+    committedTimestamp: number;
+    verifiedTimestamp?: number;
+    finalizedTimestamp?: number;
+
+    numRequestsProcessed: number;
+
+    totalNumRequestsProcessed: number;
+    totalNumDepositsProccesed: number;
+    totalNumOnchainWithdrawalsProcessed: number;
+    totalNumTradesProccesed: number;
+    totalNumOffchainWithdrawalsProcessed: number;
+    totalNumOrderCancellationsProcessed: number;
+
+    transactionHash: string;
+  }
+
+  export interface Token {
+    tokenID: number;
+    address: string;
+    enabled: boolean;
+  }
+
+  export interface Deposit {
+    blockIdx?: number;
+    requestIdx?: number;
+
+    depositIdx: number;
+    timestamp: number;
+
+    accountID: number;
+    tokenID: number;
+    amount: BN;
+    publicKeyX: string;
+    publicKeyY: string;
+
+    transactionHash: string;
+  }
+
+  export interface OnchainWithdrawal {
+    blockIdx?: number;
+    requestIdx?: number;
+
+    withdrawalIdx: number;
+    timestamp: number;
+
+    accountID: number;
+    tokenID: number;
+    amountRequested: BN;
+    amountWithdrawn?: BN;
+
+    transactionHash: string;
+  }
+
+  export interface Trade {
+    blockIdx: number;
+    requestIdx: number;
+
+    accountIdA: number;
```

Comment:
should we use M(maker) to replace A and T(taker) to replace B, not only here but also in the protocol?

Reply:
The naming with `A` and `B` predates the taker/maker decision and it would take a lot of refactoring work to change it. Naming them `takerOrder` and `makerOrder` would also work I think. It would certainly be an improvement to rename them, but it will take some time to do all the refactoring.

Reply:
ok, I'm fine without changing the names. We just need to memorize A is maker and B is taker.